addons/mail/controllers/discuss.py

```diff
@@ -107,3 +107,7 @@ def message_get_suggested_recipients(self, model, res_ids):
         except Exception:
             return {}
         return records._message_get_suggested_recipients()
+
+    @http.route('/mail/edit_message/<model("mail.message"):message>', type='json', auth='user')
+    def edit_message(self, message, body):
+        return request.env[message.model]._message_edit_content(message, body)
```

Comment:
Would say ``request.env[message.model].browse([message.res_id])._message_edit_content(message, body)`` in order to have a real recordset